packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -72,7 +77,15 @@ class ${class_name}:
         if grpc_channel is not None:
             self._stub = v2_measurement_service_pb2_grpc.MeasurementServiceStub(grpc_channel)
         self._create_file_descriptor()
-        
+        self.pin_map_id: str = ""
+        self.sites: list[int] = [0]
+
+    @cached_property
```

Comment:
I'm guessing the reason to use a cache here is that the PinMapClient construction is expensive (timely) or that we want to only ever construct one per class.; however, that renders this property as immutable...

Can the _discovery_client or the _grcp_channel_pool ever change?

Is there a reason to prefer a lazy-cached property over creating the PinMapClient during `__init__` and storing the result in an object field?

Reply:
Replaced the cached property by creating a client during `__init__` and storing the result in a private field.

Reply:
The other client objects use lazy initialization in order to defer the expensive operations (DiscoveryService.resolve_service, creating gRPC channels, and DNS lookups) until needed. I previously requested that this client do the same: https://github.com/ni/measurement-plugin-python/pull/841#discussion_r1718982257

_discovery_client and _grpc_channel_pool do not change.

I prefer lazy initialization, but not using `@cached_property`, because its locking behavior was too coarse in older Python versions.